content/post/targeted-quantiles.md

```diff
@@ -0,0 +1,160 @@
++++
+title = "Targeted Quantiles in Prometheus"
+date = "2018-03-11T19:25:53Z"
+draft = false
++++
+
+_The algorithm behind Prometheus' Summaries._
+
+## Abstract
+
+Prometheus uses a technique called Targeted Quantiles to implement the Summary metric type which is used to monitor distributions, such as latencies.
+Here, we give an overview of how the algorithm works, we answer why it is so efficient and how to configure Summaries.
+
+## Introduction
+
+A very important way of understanding how your distributed system works in production is to monitor it.
+In recent years, Prometheus has emerged as the most popular monitoring system for applications deployed on Kubernetes.
+
+Apart from the more common counters and gauges, the Prometheus client exposes the [summary](https://godoc.org/github.com/prometheus/client_golang/prometheus#Summary) metric type.
+In a go project, it might look like this:
+
+```golang
+// This is the definition of a metric which tracks the latency of an RPC request.
+latency := prometheus.NewSummary(prometheus.SummaryOpts{
+  Name:       "rpc_durations_seconds",
+  Help:       "RPC latency distributions.",
+  Objectives: map[float64]float64{0.5: 0.05, 0.9: 0.01, 0.99: 0.001},
+})
+
+// This is how you would use it somewhere in your code.
+reqDuration := rand.Float64()
+latency.Observe(reqDuration)
+```
+
+This code was extracted from the [prometheus/client-golang example](https://github.com/prometheus/client_golang/blob/master/examples/random/main.go)
+
+In the above case, the summary will give us the 99th, 90th and 50th percentile for the latency of that particular RPC request.
+
+This is what it would look like in Graphana, a popular frontend for Prometheus:
```

Comment:
frontend [framework]?

Reply:
I mean a popular interface for Prometheus

Reply:
but a good point, I'll update